commands/command_lock.go

```diff
@@ -67,6 +73,33 @@ func lockCommand(cmd *cobra.Command, args []string) {
 	}
 }
 
+type lockData struct {
+	rootDir    string
+	workingDir string
+}
+
+func computeLockData() (*lockData, error) {
+	repo, err := git.RootDir()
```

Comment:
Looking at this in the context of this PR's refactoring, I notice that this [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L790) `git rev-parse --show-toplevel`, then [performs](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L800) a Cygwin translation, and then [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L804) `tools.CanonicalizePath()`, which then starts by [running](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/tools/filetools.go#L486) another Cygwin translation before [calling](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/tools/filetools.go#L495) `tools.CanonicalizeSystemPath()`.

So, first of all, I think we should drop that first redundant [call](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L800-L803) to `tools.TranslateCygwinPath()` in `git.RootDir()`, since it'll just be run again when that calls `tools.CanonicalizePath()`.

Next, I think this function might profit by just using `cfg.LocalWorkingDir()` instead of calling `git.RootDir()`, which will always run `git rev-parse`, whereas `cfg.LocalWorkingDir()` will first [run](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L367-L368) its `loadGitDirs()` internal function, which [skips](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L380-L382) taking action if `git rev-parse` has already been used to load the various directory paths.  If the paths aren't known yet, then it [runs](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L384) `tools.GitAndRootDirs()` which [invokes](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L749) the `git rev-parse ... --show-toplevel` command and [passes](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L785) the result through `tools.CanonicalizePath()`.

(The `loadGitDirs()` function then also [uses](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L396) `tools.ResolveSymlinks()` on that path, which `git.RootDir()` and `git.GitAndRootDirs()` don't do themselves.  That may not be necessary, though, because that just [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/tools/filetools.go#L64) `tools.CanonicalizeSystemPath()` which `git.RootDir()` and `git.GitAndRootDirs()` have already done by [calling](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L785) `tools.CanonicalizePath()` which also [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/tools/filetools.go#L495) `tools.CanonicalizeSystemPath()`.  So that's yet another possible simplification, removing the `tools.ResolveSymlinks()` [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L394-L396) in `loadGitDirs()`.)

Whew ... anyway, my point in all of that is that if we use `cfg.LocalWorkingDir()` here I think we'll avoid any excess calls to `git rev-parse`, and that ties in with my comment above regarding moving the `computeLockData()` call to the top of `lockCommand()`, before `newLockClient()`, since that also [uses](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/commands/commands.go#L107) `cfg.LocalWorkingDir()` and therefore will get the benefit of this function having already run and cached the output from `git rev-parse`.  I hope that all makes sense!

Reply:
Thanks for the fixup!  I'm just going to highlight two notes from my previous comments so I don't forget that they things we could tune in another PR, but they're OK for now.
> So, first of all, I think we should drop that first redundant [call](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L800-L803) to `tools.TranslateCygwinPath()` in `git.RootDir()`, since it'll just be run again when that calls `tools.CanonicalizePath()`.

> (The `loadGitDirs()` function then also [uses](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L396) `tools.ResolveSymlinks()` on that path, which `git.RootDir()` and `git.GitAndRootDirs()` don't do themselves.  That may not be necessary, though, because that just [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/tools/filetools.go#L64) `tools.CanonicalizeSystemPath()` which `git.RootDir()` and `git.GitAndRootDirs()` have already done by [calling](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/git/git.go#L785) `tools.CanonicalizePath()` which also [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/tools/filetools.go#L495) `tools.CanonicalizeSystemPath()`.  So that's yet another possible simplification, removing the `tools.ResolveSymlinks()` [calls](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/config/config.go#L394-L396) in `loadGitDirs()`.)